### PR TITLE
Volume datasource update

### PR DESCRIPTION
**Pull request by okrause**
Volume datasource is massively outdated. This PR brings it to the currently functionality level of volume resource.
Approach taken:
1. Copy Schema from volume resource
2. Change all "required" and "optional" field into "Computed" fields, except the input parameters region and name.
3. Remove all defaults in Schema
4. Use read operation from volume resource to read the volume and populate all the fields. That saves a lot of code repetition and eliminates a lot of code divergence.

### CODE REVIEW

**Review by wenjun666 — APPROVED** [review]

File: gcp/data_source_gcp_volume.go [hunk 1]
```diff
@@ -182,7 +184,7 @@ func dataSourceGCPVolume() *schema.Resource {
 										Computed: true,
 									},
 									"has_root_access": {
-										Type:     schema.TypeBool,
+										Type:     schema.TypeString,
```
Comment (lonico) [inline gcp/data_source_gcp_volume.go:187]: why not bool?

**Review by lonico — APPROVED** [review]

File: gcp/data_source_gcp_volume.go (hunk 1 above)
Comment (okrause) [reply gcp/data_source_gcp_volume.go:187]: The schema in the datasource is a copy of the schema of the volume. In volume, it was bool in the beginning, but was changed from bool to string in https://github.com/NetApp/terraform-provider-netapp-gcp/commit/b236d0f8a92d2ff69fd16cce0c767253963dbeb3. I don't know (or forgot) the reason. @wenjun666 might know.

Comment (lonico) [reply gcp/data_source_gcp_volume.go:187]: When I look in swagger, I don't see this parameter:
```
ExportPolicy{
description:	
Export Policy for a volume

rules*	[
maxItems: 5
SimpleExportPolicyRule{Jump to definition
description:	
A Simple Export Policy Rule that defines who can access the volume and how. Only 5 rules are allowed but each rule can serve many clients and subnets.

access*	[...]
allowedClients*	[...]
nfsv3	{...}
nfsv4	{...}

Comment (okrause) [reply gcp/data_source_gcp_volume.go:187]: Are you using the latest swagger?

```bash
curl https://cloudvolumesgcp-api.netapp.com/swagger.json | jq .definitions.SimpleExportPolicyRule_v2.properties.hasRootAccess
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  326k    0  326k    0     0   408k      0 --:--:-- --:--:-- --:--:--  413k
{
  "description": "If enabled (true or on) the rule defines that no_root_squash is set, else if it is disabled (false or off) root_squash is set and user ID mapped to anonymous user 65534. The response will always be true or false and never on/off. If the value is set to null the response will always be set to the default of true.",
  "type": "string",
  "default": "true",
  "enum": [
    "true",
    "false",
    "on",
    "off"
  ],
  "x-nullable": true,
  "example": "true"
}
```

Comment (lonico) [reply gcp/data_source_gcp_volume.go:187]: Apparently not, though I'm importing the same link in https://editor.swagger.io/.   Maybe it's cached.  it shows 1.1.6

Anyway, this answers my question about string vs bool.

**Pull request merged**